galsim/shear.py

```diff
@@ -76,169 +77,279 @@ class Shear(object):
+            if 'beta' not in kwargs:
+                raise TypeError(
+                    "Shear constructor requires position angle when q is specified!")
+            beta = kwargs.pop('beta')
+            if not isinstance(beta, galsim.Angle):
+                raise TypeError(
+                    "The position angle that was supplied is not an Angle instance!")
+            q = kwargs.pop('q')
+            if q <= 0 or q > 1:
+                raise ValueError("Cannot use requested axis ratio of %f!"%q)
+            eta = -numpy.log(q)
+            self._g = self._eta2g(eta) * eta * numpy.exp(2j * beta.rad())
+
+        elif 'beta' in kwargs:
+            raise TypeError("beta provided to Shear constructor, but not g/e/eta/q")
+
+        # check for the case where there are 1 or 2 kwargs that are not valid ones for
+        # initialization a Shear
+        if kwargs:
+            raise TypeError(
+                "Shear constructor got unexpected extra argument(s): %s"%kwargs.keys())
 
     # define all the methods to get shear values
-    def getE1(self): return self._shear.getE1()
-    def getE2(self): return self._shear.getE2()
-    def getE(self): return self._shear.getE()
-    def getESq(self): return self._shear.getESq()
-    def getBeta(self): return self._shear.getBeta()
-    def getEta(self): return self._shear.getEta()
-    def getG(self): return self._shear.getG()
-    def getG1(self): return self._shear.getG1()
-    def getG2(self): return self._shear.getG2()
+    def getG1(self): 
+        """Return the g1 component of the reduced shear.
+        Note: s.getG1() is equivalent to s.g1
+        """
+        return self._g.real
+
+    def getG2(self):
+        """Return the g2 component of the reduced shear.
+        Note: s.getG2() is equivalent to s.g2
+        """
+        return self._g.imag
+
+    def getG(self): 
+        """Return the magnitude of the reduced shear |g1 + 1j * g2| = sqrt(g1**2 + g2**2)
+        Note: s.getG() is equivalent to s.g
+        """
+        return abs(self._g)
+
+    def getBeta(self):
+        """Return the position angle of the reduced shear g exp(2j beta) == g1 + 1j * g2
+        Note: s.getBeta() is equivalent to s.beta
+        """
+        return 0.5 * numpy.angle(self._g) * galsim.radians
+
+    def getShear(self):
+        """Return the reduced shear as a complex number g1 + 1j * g2
+        Note: s.getShear() is equivalent to s.shear
+        """
+        return self._g
+
+    def getE1(self):
+        """Return the e1 component of the distortion.
+        Note: s.getE1() is equivalent to s.e1
+        """
+        return self._g.real * self._g2e(abs(self._g)**2)
+
+    def getE2(self):
+        """Return the e2 component of the distortion.
+        Note: s.getE2() is equivalent to s.e2
+        """
+        return self._g.imag * self._g2e(abs(self._g)**2)
+
+    def getE(self):
+        """Return the magnitude of the distortion |e1 + i e2| = sqrt(e1**2 + e2**2)
```

Comment:
I think the reason is that for getShear() I thought it made more sense to use the python version of i, which is j.  So I wanted to point out that getShear() was g1 + 1j * g2, which is how you would build the complex object in python.  Then that idea carried over into some others like getG and getBeta.

But for these, it's probably better to just use i (like I did here), since they are all just describing the math, not python, and i is the more normal thing to use for that.